Use reference json file to define paths in building celltype refs

allyhawkins opened the pull request:
Closes #390 
Stacked on #396 

This PR updates the `build-celltype-refs.nf` workflow to use the paths defined in `scpca-refs.json` rather than re-defining them at the top of the script. Originally in #390 I was thinking I would also add a cell type specific json file, but as I worked on it that seemed less helpful. We only have a few parameters that are mostly used to define publish directories and then the other ones are paths to reference files we don't want to store in S3. So in conclusion, I didn't fully make a new cell type specific json file, but instead updated this to remove hard coding paths.

sjspielman commented on build-celltype-ref.nf:
```diff
@@ -83,6 +78,9 @@ process generate_cellassign_refs {
 workflow build_celltype_ref {
 
+  // read in json file with all reference paths
+  ref_paths = Utils.getMetaVal(file(params.ref_json), "Homo_sapiens.GRCh38.104")
```
Do you mean...
```suggestion
  ref_paths = Utils.getMetaVal(file(params.ref_json), params.celltype_organism)
```

sjspielman reviewed (commented):
I think this looks good! I didn't directly suggest this, but I wonder if we should swap `tx2gene` --> `t2g_3col_path` for a bit clearer readability here...

- `save_singler_refs` returns `t2g_3col_path` in 3rd spot in tuple (L11)
- then gets immediately consumed by `train_singler_models` which refers to that as `tx2gene`(L33 from L104)

does this make sense to you? also asking because i want to make sure i followed all the moving parts correctly, lots of scrolling..!

allyhawkins commented:
Thanks for the review @sjspielman. I addressed your comments including the use of `t2g_3col_path` throughout rather than switching variable names. This should be ready for another look.

sjspielman commented on build-celltype-ref.nf:
```diff
@@ -60,9 +56,8 @@ process generate_cellassign_refs {
   publishDir "${cellassign_ref_dir}"
```
```suggestion
  publishDir "${params.cellassign_ref_dir}"
```

sjspielman reviewed (commented):
When testing the workflow, it ran without error... but while all of the SingleR references were created as expected, the CellAssign ones weren't actually saved. I'm 99% sure my suggestion will fix this; running it through again now, we'll find out shortly!

sjspielman reviewed (approved):
Yup, that fixed it! Happy to approve now, just made sure to accept the suggestion 🚀
